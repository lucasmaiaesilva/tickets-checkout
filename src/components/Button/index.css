.button-wrapper {
  margin-top: 1rem;
}

.button-wrapper button {
  width: 100%;
  background-color: rgb(68, 142, 96);
  font-size: 1rem;
  font-size: .9rem;
  padding-top: 8px;
  padding-bottom: 8px;
  border-radius: 3px;
  color: #fff;
}

.button-wrapper button:hover {
  opacity: .9;
  cursor: pointer;
}

@media screen and (min-width: 734px) {
  .button-wrapper {
    display: flex;
  }
  .button-wrapper button {
    width: 200px;
    margin: auto;
  }
}

