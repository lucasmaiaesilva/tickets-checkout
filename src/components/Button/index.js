import React from 'react';

import './index.css';

const Button = ({ children }) => (
  <div className="button-wrapper">
    <button>
      { children }
    </button>
  </div>
);

export default Button;

