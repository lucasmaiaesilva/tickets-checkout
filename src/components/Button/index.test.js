import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './';

describe('[Component] Button', () => {
  const text = 'hello world!';
  it('render correctly', () => {
    const wrapper = shallow(<Button>{text}</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});

