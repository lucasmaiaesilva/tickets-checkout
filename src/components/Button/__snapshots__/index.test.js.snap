// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    hello world!
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button>
        hello world!
</button>,
      "className": "button-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "hello world!",
      },
      "ref": null,
      "rendered": "hello world!",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button>
          hello world!
</button>,
        "className": "button-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello world!",
        },
        "ref": null,
        "rendered": "hello world!",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
